lib/Sema/CSDiag.cpp

```diff
@@ -3127,12 +3133,6 @@ void ConstraintSystem::diagnoseAssignmentFailure(Expr *dest, Type destTy,
                                                  SourceLoc equalLoc) {
   auto &TC = getTypeChecker();
 
-  // Diagnose obvious assignments to literals.
-  if (isa<LiteralExpr>(dest->getValueProvidingExpr())) {
```

Comment:
I think it's still useful to diagnose stuff like `7 = 42` which this error message, but unfortunately I haven't found a test for that in `test/Sema/immutability.swift`...

Reply:
< I think it's still useful to diagnose stuff like 7 = 42
Of course, I deleted the check here because I noticed it is checked *anyways* and always in the function that calls this one. Would you like me to add a test for that?